client/web/src/repo/commit/RepositoryCommitPage.tsx

```diff
@@ -71,8 +60,70 @@ export const RepositoryCommitPage: React.FunctionComponent<RepositoryCommitPageP
         [data]
     )
 
+    return (
+        <RepositoryRevisionNodes
+            error={error}
+            loading={loading}
+            revspec={params.revspec}
+            changelistID=""
+            commit={commit}
+            {...props}
+        />
+    )
+}
+
+/** Displays a changelist. */
+export const RepositoryChangelistPage: React.FunctionComponent<RepositoryCommitPageProps> = props => {
+    const params = useParams<{ changelistID: string }>()
+
+    if (!params.changelistID) {
+        throw new Error('Missing `changelistID` param! It must be set.')
+    }
+
+    const { data, error, loading } = useQuery<RepositoryChangelistResult, RepositoryChangelistVariables>(
+        CHANGELIST_QUERY,
+        {
+            variables: {
+                repo: props.repo.id,
+                changelistID: params.changelistID,
+            },
+        }
+    )
+
+    const commit = useMemo(
+        () => (data?.node && data?.node?.__typename === 'Repository' ? data?.node?.changelist?.commit : undefined),
+        [data]
+    )
+
+    return (
+        <RepositoryRevisionNodes
+            error={error}
+            loading={loading}
+            changelistID={params.changelistID}
+            revspec=""
+            commit={commit}
+            {...props}
+        />
+    )
+}
+
+interface RepositoryRevisionNodesProps extends TelemetryProps, PlatformContextProps, SettingsCascadeProps {
+    error: ApolloError | undefined
+    loading: boolean
+
+    revspec: String
+    changelistID: String
+
+    repo: RepositoryFields
+    commit: GitCommitFields | undefined | null
+    onDidUpdateExternalLinks: (externalLinks: ExternalLinkFields[] | undefined) => void
+}
+
+const RepositoryRevisionNodes: React.FunctionComponent<RepositoryRevisionNodesProps> = props => {
     const [diffMode, setDiffMode] = useTemporarySetting('repo.commitPage.diffMode', 'unified')
 
+    const { error, loading, commit, repo } = { ...props }
```

Comment:
```suggestion
    const { error, loading, commit, repo } = props
```

Think this should work?